src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3109,6 +3158,7 @@ internal Buffer(IEnumerable<TElement> source)
                     count++;
                 }
             }
+
             this.items = items;
```

Comment:
That's a good question. I'd be interested to know how implementation dependencies affect compatibility in general. /cc @ericstj